common/flagdproxy/flagdproxy.go

```diff
@@ -178,31 +192,46 @@ func (f *FlagdProxyHandler) newFlagdProxyManifest(ownerReferences []metav1.Owner
 	}
 }
 
-func (f *FlagdProxyHandler) doesFlagdProxyExist(ctx context.Context) (bool, error) {
+func (f *FlagdProxyHandler) doesFlagdProxyExist(ctx context.Context) (bool, *appsV1.Deployment, error) {
 	d := &appsV1.Deployment{}
 	err := f.Client.Get(ctx, client.ObjectKey{Name: FlagdProxyDeploymentName, Namespace: f.config.Namespace}, d)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// does not exist, is not ready, no error
-			return false, nil
+			return false, nil, nil
 		}
 		// does not exist, is not ready, is in error
-		return false, err
+		return false, nil, err
 	}
-	// exists, at least one replica ready, no error
-	return true, nil
+	return true, d, nil
 }
 
-func (f *FlagdProxyHandler) getOwnerReference(ctx context.Context) (metav1.OwnerReference, error) {
+func (f *FlagdProxyHandler) isFlagdProxyUpToDate(old, new *appsV1.Deployment) bool {
+	return reflect.DeepEqual(old.Spec, new.Spec)
```

Comment:
it would be interesting in the future to allow fix releases (basically, pin only major.minor) of flagd w/o the need of a new OFO version

Reply:
yeah, I thought about this too. To compare versions, we need to go deep into the deployment content (to the container level). Besides, I think we are safe with renovate and releases to make sure this fails only when there're upgrades.

Reply:
But there are also other use-cases when we should update, let's imagine a user changes `FeatureFlagSource`, or env variable value for OFO Deployment (let's say `management-port`) -> this needs to change the deployment of flagd-proxy and adapt it according to the configuration. Therefore I think it's not enough to check for the image, as the flagd-proxy deployment needs to be adapted even if other configuration changes. This was the reason I went for the comparison of the whole spec.

Reply:
I think there is a misunderstanding here @odubajDT :) 
What I was thinking for the future, after flagd is at v1.x.y, is to allow old and new `.spec` fields] to differ only in the `image` field for the `tag` part. Since flagd follows semantic versioning, we would have a version schema `x.y.z` and we can check that new`z` is >= than old `z`. This way, users can patch their flagd-proxy manually to have a quick fix if smth is off without waiting for the related OFO release :) 
I hope this clarifies my comment :)